test/uk/gov/hmrc/cipphonenumberverification/services/NotificationServiceSpec.scala

```diff
@@ -16,137 +16,208 @@
         .returns(Future.successful(Right(httpResponse)))
 
       val result = service.status(notificationId)
+
       status(result) shouldBe 200
       (contentAsJson(result) \ "code").as[Int] shouldBe 108
       (contentAsJson(result) \ "message").as[String] shouldBe "There is a problem with the notification vendor"
+      verify(mockGovNotifyUtils).extractPasscodeFromGovNotifyBody(expectedGovNotifyResponseBody)
+      val expectedVerificationDeliveryResultRequestAuditEvent: VerificationDeliveryResultRequestAuditEvent = VerificationDeliveryResultRequestAuditEvent(expectedPhoneNumber, passcode, expectedNotificationid, "technical-failure")
+      mockAuditService.sendExplicitAuditEvent("PhoneNumberVerificationDeliveryResultRequest", expectedVerificationDeliveryResultRequestAuditEvent) was called
     }
 
-    "return NotificationStatus when notification id not found" in {
+    "return NotificationStatus when notification id not found" in new SetUp {
       val httpResponse = UpstreamErrorResponse("", Status.NOT_FOUND)
       mockGovUkConnector.notificationStatus(notificationId)
         .returns(Future.successful(Left(httpResponse)))
 
       val result = service.status(notificationId)
+
       status(result) shouldBe 404
       (contentAsJson(result) \ "code").as[String] shouldBe "NOTIFICATION_NOT_FOUND"
       (contentAsJson(result) \ "message").as[String] shouldBe "Notification ID not found"
+      verifyNoInteractions(mockGovNotifyUtils)
+      val expectedVerificationDeliveryResultRequestAuditEvent: VerificationDeliveryResultRequestAuditEvent = VerificationDeliveryResultRequestAuditEvent("NOTIFICATION_NOT_FOUND", "NOTIFICATION_NOT_FOUND", notificationId, "NOTIFICATION_NOT_FOUND")
+      mockAuditService.sendExplicitAuditEvent("PhoneNumberVerificationDeliveryResultRequest", expectedVerificationDeliveryResultRequestAuditEvent) was called
     }
+
+  }
+
+  trait SetUp {
+    implicit val writes: OWrites[GovUkNotificationStatusResponse] = Json.writes[GovUkNotificationStatusResponse]
+    implicit val hc: HeaderCarrier = HeaderCarrier()
+    val mockAuditService = mock[AuditService]
+    val mockGovUkConnector = mock[GovUkConnector]
+    val mockGovNotifyUtils = mock[GovNotifyUtils]
+    val service = new NotificationService(mockGovNotifyUtils, mockAuditService, mockGovUkConnector)
+    val notificationId = "740e5834-3a29-46b4-9a6f-16142fde533a"
+    val expectedNotificationid = "740e5834-3a29-46b4-9a6f-16142fde533a"
+    val passcode = "XYZABC"
+    val expectedPhoneNumber = "+447900900123"
+    val expectedGovNotifyResponseBody = "CIP Phone Number Verification Service: theTaxService needs to verify your telephone number.\n    Your telephone number verification code is ABCDEF.\n    Use this code within 10 minutes to verify your telephone number."
+    when(mockGovNotifyUtils.extractPasscodeFromGovNotifyBody(any[String])).thenReturn(passcode)
   }
+
+  private def buildGovNotifyResponseWithStatus(status: String): JsValue = {
```

Comment:
not sure why we need this. All we care about is ensuring that we get back an object from the connector with a 'status' field, which we are already checking in the existing unit test

Reply:
- The connector returns a HttpResponse or an UpstreamErrorResponse.
- IF the connector returned a GovUkNotificationStatusResponse then GovUkNotificationStatusResponse could be used as the input to the unit tests for the NotificationServiceSpec.  But it doesn't.  
- The NotificationService is doing the deserialising of the response into a GovUkNotificationStatusResponse.  
- Therefore the tests for NotificationService should take the raw http response as input.
- This enables the deserialising of the code from the httpResponse into the the GovUkNotificationStatusResponse object to be tested.

Reply:
The tests currently use
Json.toJson(govUkNotificationStatus)
to setup the mock. This essentially gives us a raw http response object by converting the GovUkNotificationStatusResponse into a JsValue

Reply:
Exactly, this is better.